core/src/main/java/io/questdb/cairo/O3PartitionJob.java

```diff
@@ -970,6 +1086,183 @@ private static long getDedupRowsWithAdditionalKeys(
+                    dstFixMem.extend(dstFixSize);
+                    MemoryCARWImpl dstVarMem = new MemoryCARWImpl(PAGE_SIZE, Integer.MAX_VALUE, MemoryTag.NATIVE_O3);
+                    dstVarMem.extend(dstVarSize);
+
+                    mergedMemory.add(dstFixMem);
+                    mergedMemory.add(dstVarMem);
+
+                    O3CopyJob.mergeCopy(
+                            columnType,
+                            timestampMergeIndexAddr,
+                            mergeRowCount,
+                            columnAuxPtr,
+                            columnDataPtr,
+                            srcOooFixAddr,
+                            srcOooVarAddr,
+                            dstFixMem.addressOf(0),
+                            dstVarMem.addressOf(0),
+                            0
+                    );
+
+                    partitionDescriptor.addColumn(
+                            columnName,
+                            columnType,
+                            columnId,
+                            0,
+                            dstVarMem.addressOf(0),
+                            dstVarSize,
+                            dstFixMem.addressOf(0),
+                            dstFixSize,
+                            0,
+                            0
+                    );
+                } else {
+                    final long srcOooFixAddr = oooMem1.addressOf(0);
+                    long dstFixSize = mergeRowCount * ColumnType.sizeOf(columnType);
+
+                    MemoryCARWImpl dstFixMem = new MemoryCARWImpl(PAGE_SIZE, Integer.MAX_VALUE, MemoryTag.NATIVE_O3);
+                    dstFixMem.extend(dstFixSize);
+                    mergedMemory.add(dstFixMem);
+
+                    // Merge column data
+                    O3CopyJob.mergeCopy(
+                            notTheTimestamp ? columnType : ColumnType.setDesignatedTimestampBit(columnType, true),
+                            timestampMergeIndexAddr,
+                            mergeRowCount,
+                            columnDataPtr,
+                            0,
+                            srcOooFixAddr,
+                            0,
+                            dstFixMem.addressOf(0),
+                            0,
+                            0
+                    );
+
+                    partitionDescriptor.addColumn(
+                            columnName,
+                            columnType,
+                            columnId,
+                            0,
+                            dstFixMem.addressOf(0),
+                            dstFixSize,
+                            0,
+                            0,
+                            0,
+                            0
+                    );
+                }
+            }
+            PartitionEncoder.update(partitionDescriptor, (short) rowGroup, path);
```

Comment:
We already have the parquet file descriptor open for reading in the `file` variable, it would be better to use it instead of passing the path.

Reply:
native rust implementation is the owner of file descriptor,  PartitionEncoder shold open a new one to use